default.nix

```diff
@@ -1,37 +1,2 @@
-scope@{ pkgs ? import <nixpkgs> { } }:
-
-pkgs.buildEnv {
-  name = "console-env";
-  paths = with pkgs;
-    [
-      git
-      bash
-      direnv
-      binutils
-      stdenv
-      bashInteractive
-      docker
-      cacert
-      gcc
-      cmake
-      rustup
-      pkg-config
-      openssl
-      protobuf
-      docker
-      (glibcLocales.override { locales = [ "en_US.UTF-8" ]; })
-    ] ++ lib.optional stdenv.isDarwin [ Security libiconv ];
-  passthru = with pkgs; {
-    PROTOC = "${protobuf}/bin/protoc";
-    PROTOC_INCLUDE = "${protobuf}/include";
-    LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
-    LC_ALL = "en_US.UTF-8";
-    OPENSSL_DIR = "${openssl.dev}";
-    OPENSSL_LIB_DIR = "${openssl.out}/lib";
-    SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
-    GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
-    CURL_CA_BUNDLE = "${cacert}/etc/ca-bundle.crt";
-    CARGO_TERM_COLOR = "always";
-    RUST_BACKTRACE = "full";
-  };
-}
+{ pkgs ? import <nixpkgs> { } }:
```

Comment:
hmm, there are now enough nix configs in the repository that i wonder if we should stick them in `console/nix/` instead of having them all in `console/`? not a hard blocker though...

Reply:
Kinda depends on how people are going to be using `nix`. I think the `flake.{nix,lock}` are the only things that actually *have* to live at the repo root. `shell.nix` and `default.nix` might be happier at the root if people are going to be invoking `nix-build` and `nix-shell` manually, since the `nix-*` commands will look in their current directory for their respective files first. Otherwise, you'd have to give them a path to their file.

If `.envrc` is going to be the main nix entrypoint for most, then everything but the flake can definitely be relocated.

Reply:
:woman_shrugging: we can figure that out later, i think leaving them all in the root is fine for now.

Reply:
Eh, I went ahead and moved them. I figure since the only thing that was here before was the `.envrc` and `shell.nix`, it's pretty unlikely that anyone's existing workflow will be significantly disrupted. I don't know anyone who manually uses `nix-shell` anymore, and there was nothing to `nix-build` to begin with :grin: